tests/ResolverTests.hs

```diff
@@ -37,6 +37,33 @@ tHandler :: Handler TMonad T
 tHandler =
   pure $ (pure 10) :<> (\tArg -> pure tArg) :<> (pure . (*2))
 
+
+-- https://github.com/jml/graphql-api/issues/119
+-- Maybe X didn't descent into its argument. Now it does.
+type Query = Object "Query" '[]
+  '[ Argument "id" Text :> Field "test" (Maybe Foo) ]
+
+type Foo = Object "Foo" '[]
+  '[ Field "name" Text ]
+
+data ServerFoo = ServerFoo
+  { name :: Text
+  } deriving (Eq, Show)
+
+lookupFoo :: Text -> IO (Maybe ServerFoo)
+lookupFoo _ = pure $ Just (ServerFoo "Mort")
+
+viewFoo :: ServerFoo -> Handler IO Foo
+viewFoo ServerFoo { name=name } = pure $ pure $ name
+
+handler :: Handler IO Query
+handler = pure $ \fooId -> do
+  foo <- lookupFoo fooId
+  -- note that fmap maps over the Maybe, so we still need
+  -- have to wrap the result in a pure.
+  sequence $ fmap (pure . viewFoo) foo
```

Comment:
A little bit weird that this isn't executed. I guess we're relying on the compiler?

Reply:
fixed